src/pages/main-page/MainPage.tsx

```diff
@@ -6,55 +6,76 @@ import Select from '../../components/blocks/select/Select';
 import { CityPropsType, PlaceCardType } from '../../components/blocks/place-card/types';
 import { MainPropsType } from './types';
 
-export default function MainPage({placesAmount, places, cities, filters }: MainPropsType): JSX.Element {
+export default function MainPage({ places, cities, filters }: MainPropsType): JSX.Element {
   const [activeCardId, setActiveCardId] = useState<PlaceCardType['id']>(null);
-  const [activeCity, ] = useState<CityPropsType>(cities[0]);
-  const [activeCityName, setActiveCityName] = useState<CityPropsType['name']>(cities[0].name);
+  const [activeCity,] = useState<CityPropsType>(cities[0]);
+  const [activeCityName, setActiveCityName] = useState<CityPropsType['name']>(cities[0]?.name);
+  const [isSelected, setIsSelected] = useState(filters[0]);
+  const filteredPlaces = places.filter((place) => activeCityName === place.city.name);
 
-  function onCardHover(placeId: PlaceCardType['id']): void {
+  const onCardHover = (placeId: PlaceCardType['id']): void => {
     places.some((place) => {
       if (place.id === placeId) {
         setActiveCardId(placeId);
       }
     });
-  }
+  };
 
-  function onCityItemClick(cityName: CityPropsType['name']): void {
+  const onCityItemClick = (cityName: CityPropsType['name']): void => {
     cities.find((el) => {
       if (el.name === cityName) {
         setActiveCityName(cityName);
       }
     });
-  }
+  };
+
+  const onSelectItemClick = (selectName: string) => {
+    filters.find((filter) => {
+      if (filter === selectName) {
+        setIsSelected(filter);
+      }
+    });
+  };
+
+  const onSelectItemClickForFilters = (cards: Array<PlaceCardType>) => {
+    const copyPlaces: Array<PlaceCardType> = JSON.parse(JSON.stringify(cards));
```

Comment:
вот здесь чтобы типы не ругались, можно вот так сделать:
```suggestion
    const copyPlaces: Array<PlaceCardType> = JSON.parse(JSON.stringify(cards)) as Array<PlaceCardType>;
```

Reply:
готово